pkg/api/validation/validation.go

```diff
@@ -1907,7 +1905,10 @@ func ValidateTolerationsInPodAnnotations(annotations map[string]string, fldPath
 	return allErrs
 }
 
-func validateSeccompProfile(p string, fldPath *field.Path) field.ErrorList {
+func ValidateSeccompProfile(p string, fldPath *field.Path) field.ErrorList {
+	if p == "" {
```

Comment:
In this case we're dealing with the internal object and the field is not a
pointer. In a general case, empty means use the default (psp or runtime)

On Thursday, July 28, 2016, Paul Morie <notifications@github.com> wrote:

> In pkg/api/validation/validation.go
> <https://github.com/kubernetes/kubernetes/pull/28300#discussion_r72672467>
> :
>
> > @@ -1907,7 +1905,10 @@ func ValidateTolerationsInPodAnnotations(annotations map[string]string, fldPath
> >  	return allErrs
> >  }
> >
> > -func validateSeccompProfile(p string, fldPath *field.Path) field.ErrorList {
> > +func ValidateSeccompProfile(p string, fldPath *field.Path) field.ErrorList {
> > +	if p == "" {
>
> What's the use-case for the annotation being present, but having no value?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/28300/files/64110ad8fcd2f9dcb5bf4ef6d982e56ec0aa784e#r72672467>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AE_PUwcpI_FXqYkYtBjOwwHIVueF1qylks5qaPDEgaJpZM4JCbSe>
> .
>